### PR TITLE
[DOC] use short ref to nilearn functions and classes in the examples

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

This PR suggest to change:

- replaces ``:func:`nilearn.`` by ``:func:`~nilearn.`` 
- replaces ``:class:`nilearn.`` by ``:class:`~nilearn.`` 

This should shorten the render of 'objects' in the example from `nilearn.surface.SurfaceImage` to `SurfaceImage`: so easier to read.

BUT I am not 100% sure this is a good idea as sometimes it may be good for user to know where to import a specific function from by seeing it in the doc rather than having to click on a link to know that.

Opinions welcome.

@man-shu @bthirion @emdupre

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
LGTM. Is the build doc failure related to that PR ?

**Comment by Remi-Gau** [comment]
> LGTM. Is the build doc failure related to that PR ?

oops no it is related: let me fix

**Comment by Remi-Gau** [comment]
doc rendering blocked by https://github.com/nilearn/nilearn/pull/4911

File: examples/00_tutorials/plot_3d_and_4d_niimg.py [hunk 1]
```diff
@@ -126,7 +126,7 @@
 # given in the documentation section: :ref:`loading_data`.
 #
 # Functions accept either 3D or 4D images, and we need to use on the one
-# hand :func:`nilearn.image.index_img` or :func:`nilearn.image.iter_img`
+# hand :func:`~nilearn.image.index_img` or :func:`~nilearn.image.iter_img`
```
Comment (bthirion) [inline examples/00_tutorials/plot_3d_and_4d_niimg.py:129]: Should we also create a link on line 93 of this example ?

**Review by bthirion — COMMENTED** [review]
LGTM overall. Only potential issue is that some occurrences may be missed.

**Review by man-shu — APPROVED** [review]
LGTM!

**Pull request merged**